Firestore/core/src/firebase/firestore/core/sync_engine.cc

```diff
@@ -106,13 +108,12 @@ TargetId SyncEngine::Listen(Query query) {
 
 ViewSnapshot SyncEngine::InitializeViewAndComputeSnapshot(
     const local::QueryData& query_data) {
-  DocumentMap docs = local_store_->ExecuteQuery(query_data.query());
-  DocumentKeySet remote_keys =
-      local_store_->GetRemoteDocumentKeys(query_data.target_id());
+  QueryResult query_result = local_store_->ExecuteQuery(
```

Comment:
Port of https://github.com/firebase/firebase-android-sdk/blob/d8ff6a54d12fd676a31c2c331f09fe0a7d033202/firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java#L191